README.md

```diff
@@ -212,48 +232,50 @@ At this time, conversion rules only supports adding or removing columns from
 the schemas, there are no type conversions performed, nor ways to rename
 columns, etc... More advanced conversion rules may be added in the future.
 
-### Sorting Row Groups: [parquet.Buffer](https://pkg.go.dev/github.com/segmentio/parquet-go#Buffer)
+### Sorting Row Groups: [parquet.GenericBuffer[T]](https://pkg.go.dev/github.com/segmentio/parquet-go#Buffer)
 
-The `parquet.Writer` type is optimized for minimal memory usage, keeping the
-order rows unchanged and flushing pages as soon as they are filled.
+The `parquet.GenericWriter[T]` type is optimized for minimal memory usage,
+keeping the order rows unchanged and flushing pages as soon as they are filled.
 
 Parquet supports expressing columns by which rows are sorted through the
 declaration of _sorting columns_ on row groups. Sorting row groups requires
 buffering all rows before ordering and writing them to a parquet file.
 
 To help with those use cases, the `segmentio/parquet-go` package exposes the
-`parquet.Buffer` type which acts as a buffer of rows and implements
+`parquet.GenericBuffer[T]` type which acts as a buffer of rows and implements
 `sort.Interface` to allow applications to sort rows prior to writing them
 to a file.
 
 The columns that rows are ordered by are configured when creating
-`parquet.Buffer` instances using the `parquet.SortingColumns` function to
-construct row group options configuring the buffer. The type of parquet columns
-defines how values are compared, see [Parquet Logical Types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md)
+`parquet.GenericBuffer[T]` instances using the `parquet.SortingColumns` function
+to construct row group options configuring the buffer. The type of parquet
+columns defines how values are compared, see [Parquet Logical Types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md)
 for details.
 
 When written to a file, the buffer is materialized into a single row group with
 the declared sorting columns. After being written, buffers can be reused by
 calling their `Reset` method.
 
-The following example shows how to use a `parquet.Buffer` to order rows written
-to a parquet file:
+The following example shows how to use a `parquet.GenericBuffer[T]` to order rows
+written to a parquet file:
 
 ```go
-buffer := parquet.NewBuffer(
+buffer := parquet.NewGenericBuffer[RowType](
```

Comment:
Is the user supposed to define RowType? 

If so I'd maybe define it as part of the example - it's better to be explicit than implicit for docs, and to show the relationship between the objects.

Reply:
Yes, `RowType` is whatever type the application wants to use since it's a generic.

Which do you think would be better:

- define `RowType` once in the first example
- define it every time (maybe fills the example with a lot of redundant and not very relevant info)
- use a more generic name like `[T]` of `[...]` to specify that this part of the code is not relevant to the example